### ISSUE TITLE
Snapshots cause storybook to fail.

### ISSUE THREAD

**Issue by dimitrisfasoulas** [storybooks/storybook-addon-jest#8]
Adding a jest snapshot will cause storybook to fail with:

> Result.js:235 Uncaught TypeError: patterns.reduce(...).split(...).join(...).split(...).reduce(...).map(...).reduce(...).map(...).reduce(...).reduce(...).reduce(...).reduce(...).reduce(...).reduce is not a function
>     at Message (Result.js:235)

I guess the formatting logic of the error message is not accounting for snapshots. I see on result.js (which is not on the repo ??) for example:

`endTrigger: e => typeof e === 'string' && e.match('Expected ')`

The error message of a failed snapshot though is differently formatted:

> ● Description:  › renders correctly

    expect(value).toMatchSnapshot()

    Received value does not match stored snapshot 1.

    - Snapshot
    + Received

    @@ -4,8 +4,8 @@
        </li>
        <li>
          bar
        </li>
        <li>
    -     foo
    +     bar
        </li>
      </ul>

      20 | 			.create(<List items={['foo', 'bar', 'bar']} />)
      21 | 			.toJSON();
    > 22 | 		expect(tree).toMatchSnapshot();
      23 | 	});
      24 | });
      25 |

      at Object.<anonymous> (tests/list.test.js:22:16)

no __Expected__ in the error output.

Packages versions: 
```
"@storybook/addon-jest": "^3.3.9",
    "@storybook/react": "^3.3.9",
    "jest": "^22.1.2",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "react-test-renderer": "^16.2.0"
```

I am attaching a result json file that will cause the error.
[jest-test-results.txt](https://github.com/storybooks/storybook-addon-jest/files/1642303/jest-test-results.txt)

**Comment by felixjung** [comment]
For me, when snapshots tests break, storybook will fail. After updating snapshots, it works again.

**Comment by dimitrisfasoulas** [comment]
Yes, storybook fails. I think it would be better if instead, we got an error message on the exact test that failed, than crashing the whole of storybook
